### PR TITLE
CLOUDTRUST-2905: avoid double-signature on the logout response in artifact binding

### PR DESCRIPTION
**Pull request by lagess**

### CODE REVIEW

File: services/src/main/java/org/keycloak/protocol/saml/SamlService.java [hunk 1]
```diff
@@ -553,7 +553,8 @@ protected Response logoutRequest(LogoutRequestType logoutRequest, ClientModel cl
 
                     //artifact binding state must be attached to the user session upon logout, as authenticated session
                     //no longer exists when the LogoutResponse message is sent
-                    if ("true".equals(clientSession.getNote(JBossSAMLURIConstants.SAML_HTTP_ARTIFACT_BINDING.get()))){
+                    if ("true".equals(clientSession.getNote(JBossSAMLURIConstants.SAML_HTTP_ARTIFACT_BINDING.get()))
+                            && samlClient.useArtifactBindingLogout()) {
                         userSession.setNote(JBossSAMLURIConstants.SAML_HTTP_ARTIFACT_BINDING.get(), "true");
```
Comment (fperot74) [inline services/src/main/java/org/keycloak/protocol/saml/SamlService.java:558]: So if SamlArtifactBindingLogout is disabled, you don't set the user session note... And what happens? It acts as a standard saml or it does nothing? (I guess it is the second option)

**Review by fperot74 — APPROVED** [review]

File: services/src/main/java/org/keycloak/protocol/saml/SamlService.java (hunk 1 above)
Comment (lagess) [reply services/src/main/java/org/keycloak/protocol/saml/SamlService.java:558]: It does nothing particular, thus it falls back on the standard use case --> use POST / Redirect binding according to the client configuration.
Cf. https://github.com/cloudtrust/keycloak/blob/a20f3b726a7f815f14cdf11e4aea1581dfe453fe/services/src/main/java/org/keycloak/protocol/saml/SamlProtocol.java#L609

File: services/src/main/java/org/keycloak/protocol/saml/SamlProtocol.java [hunk 2]
```diff
@@ -860,8 +860,10 @@ protected Response buildLogoutArtifactResponse(UserSessionModel userSession,
             throws ConfigurationException, ProcessingException {
 
         try {
+            // performs necessary signatures and encryption on the logout response element
+            bindingBuilder.postBinding(samlDocument);
+            // wrap the logout response inside an artifact response element
             Document artifactResponse = SamlProtocolUtils.buildArtifactResponse(samlDocument);
-            bindingBuilder.postBinding(artifactResponse); //this step performs necessary signatures and encryption
```
Comment (harture) [inline services/src/main/java/org/keycloak/protocol/saml/SamlProtocol.java:864]: c'est donc ce point qui faisait la double signature ?

**Review by harture — APPROVED** [review]

File: services/src/main/java/org/keycloak/protocol/saml/SamlProtocol.java (hunk 2 above)
Comment (lagess) [reply services/src/main/java/org/keycloak/protocol/saml/SamlProtocol.java:864]: Yes! At least it is doing an additional signature. And there was already one.

**Pull request merged**